### PR TITLE
wtmcu: add initial version 0.1.0

### PR DESCRIPTION
**Pull request by evgenyz**
Maintainer: me
Compile tested: mvebu (cortext-a9), OpenWrt SNAPSHOT, r9945-bc85640cdc
Run tested: mvebu (cortext-a9), OpenWrt SNAPSHOT, r9945-bc85640cdc

Description:
Weltrend MCU control facilities. This is a supporting package for wdmc (WD My Cloud) device(es) (openwrt/openwrt#2040).

### CODE REVIEW

File: utils/wtmcu/Makefile [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+	This package provides facilities to control Weltrend MCU.
+endef
+
+define Package/wtmcu/conffiles
+	/etc/config/wtmcu
+endef
+
+TARGET_CFLAGS+=-std=gnu99
+
+#define Build/Prepare
```
Comment (neheb) [inline utils/wtmcu/Makefile]: Testing artifact?

File: utils/wtmcu/Makefile [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+define Package/wtmcu/description
+	This package provides facilities to control Weltrend MCU.
+endef
+
+define Package/wtmcu/conffiles
+	/etc/config/wtmcu
+endef
+
+TARGET_CFLAGS+=-std=gnu99
```
Comment (neheb) [inline utils/wtmcu/Makefile]: This should not be needed.

Comment (diizzyy) [reply utils/wtmcu/Makefile]: This may be needed depending on version of compiler

File: utils/wtmcu/Makefile [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+PKG_SOURCE_URL:=https://gitlab.com/evgenyz/$(PKG_NAME)/-/archive/v$(PKG_VERSION)
+PKG_HASH:=557f8b6a9771dcee094692da4480274695ec2af53871a3ddea069ee94e207509
+
+PKG_LICENSE:=BSD-2-Clause
+PKG_MAINTAINER:=Evgeny Kolesnikov <evgenyz@gmail.com>
+
+PKG_BUILD_PARALLEL:=1
+PKG_BUILD_DIR:=$(BUILD_DIR)/$(PKG_NAME)-v$(PKG_VERSION)
+
+CMAKE_INSTALL:=1
```
Comment (diizzyy) [inline utils/wtmcu/Makefile]: Remove

File: utils/wtmcu/Makefile [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+include $(INCLUDE_DIR)/package.mk
+include $(INCLUDE_DIR)/cmake.mk
+
+define Package/wtmcu
+	SECTION:=utils
+	CATEGORY:=Utilities
+	TITLE:=Weltrend MCU control daemon and console tool
+	MENU:=1
```
Comment (diizzyy) [inline utils/wtmcu/Makefile]: Why is this needed?

File: utils/wtmcu/files/wtmcud.init [hunk 5]
```diff
@@ -0,0 +1,31 @@
+#!/bin/sh /etc/rc.common
+
+START=10
```
Comment (diizzyy) [inline utils/wtmcu/files/wtmcud.init]: Why does it need to start this early?
Preferably set 99 instead as it's not system critical

Comment (evgenyz) [reply utils/wtmcu/files/wtmcud.init]: It is kind of system critical. MCU is in charge of controlling fan, temperature, power for peripherals, LEDs, WOL, APR and battery-backed clock. I'd like it to be available as early as possible as it provides ubus service to control all of this.

Comment (diizzyy) [reply utils/wtmcu/files/wtmcud.init]: Sure, but to my knowledge ubus depends on rpcd which is started later on?
https://github.com/openwrt/openwrt/blob/master/package/system/rpcd/files/rpcd.init#L3

Comment (evgenyz) [reply utils/wtmcu/files/wtmcud.init]: Well, yes and no. If your service is a plugin for rpcd (either one-off or library) you have to wait for rpcd. But wtmcud is a generic ubus service, which connects directly and does not require rpcd. Ubus is a part of procd and is already available just before the init stage.

File: utils/wtmcu/Makefile (hunk 2 above)
Comment (neheb) [reply utils/wtmcu/Makefile]: What I mean is that the package should be compatible with all versions. If not, this is a bug that should be fixed.

Comment (evgenyz) [reply utils/wtmcu/Makefile]: Well, current toolchain used for mvebu target does indeed defaults to C standard which is fresh enough to drop this flag. But, irrelevant to this package, what do you mean by "compatible with all versions"? This flag just asks for a particular C standard revision. If gcc does not support it, the flag would do nothing, it won't compile. Are you saying that package should only be compatible with what is default for currently used gcc version?

**Comment by evgenyz** [comment]
For now, the package was moved into the base openwrt/openwrt#2040, with all the fixes you proposed. No idea if it will stay there or be back (or be ditched altogether), but I'll close this PR for now.

File: utils/wtmcu/Makefile (hunk 2 above)
Comment (diizzyy) [reply utils/wtmcu/Makefile]: @neheb 
No it's not, default value may (will) change depending on GCC/compiler version. Blantantly ignoring this fact will however introduce breakage and possibly even bugs.
